examples/energy-management-app/energy-management-common/water-heater/src/WhmDelegateImpl.cpp

```diff
@@ -273,61 +300,77 @@ Status WaterHeaterManagementDelegate::HandleCancelBoost()
  *
  *********************************************************************************/
 
+uint16_t WaterHeaterManagementDelegate::GetActiveTargetWaterTemperature() const
+{
+    // Determine the target temperature. If a boost command is in progress and has a mBoostTemporarySetpoint value use that as the
+    // target temperature.
+    // Note, in practise the actual heating is likely to be controlled by the thermostat's occupiedHeatingSetpoint most of the
+    // time, and the TemporarySetpoint (if not null) would be overiding the thermostat's occupiedHeatingSetpoint.
+    // However, this code doesn't rely upon the thermostat cluster.
+    uint16_t targetTemperature = (mBoostState == BoostStateEnum::kActive && mBoostTemporarySetpoint.HasValue())
+        ? static_cast<uint16_t>(mBoostTemporarySetpoint.Value())
+        : mTargetWaterTemperature;
+
+    return targetTemperature;
+}
+
+uint8_t WaterHeaterManagementDelegate::CalculateTankPercentage() const
+{
+    int32_t tankPercentage = 100 * (static_cast<int32_t>(mWaterTemperature) - static_cast<int32_t>(mColdWaterTemperature)) /
+        (static_cast<int32_t>(GetActiveTargetWaterTemperature()) - static_cast<int32_t>(mColdWaterTemperature));
```

Comment:
If GetActiveTargetWaterTemperature() == mColdWaterTemperature this will generally speaking crash, right? What guarantees this can't happen?

Reply:
Added a check on the value of the divisor.